plugins/gradle/tooling-extension-impl/src/org/jetbrains/plugins/gradle/tooling/internal/ExtraModelBuilder.java

```diff
@@ -97,6 +111,8 @@ public Object buildAll(String modelName, Project project) {
             if (service instanceof AbstractModelBuilderService) {
               return ((AbstractModelBuilderService)service).buildAll(modelName, project, myModelBuilderContext);
             }
+            else if (service instanceof ModelBuilderService.Parametrized)
```

Comment:
we can avoid additional `instanceof` and new interface `ModelBuilderService.Parametrized` using `org.jetbrains.plugins.gradle.tooling.ModelBuilderContext`

Reply:
Do you mean I should add a new method to `ModelBuilderContext` instead?

Reply:
Yep, I'd like to deprecate `ModelBuilderService#buildAll(String modelName, Project project)` method in favour of `ModelBuilderService#buildAll(String modelName, Project project, ModelBuilderContext context)`.
At some moment in future when minimal language level of TAPI modules will be set to 1.8 and we could use "default" interface methods instead of abstract class.

Reply:
`AbstractModelBuilderService` is a class so I cannot just implement it in one Kotlin model builder which is derived fro their abstract model builder which implements `ModelBuilderService` directly. Since we do not have default methods right now, I can add an `Ex` interface with this method, make `AbstractModelBuilderService` implement it and then in the Kotlin plugin implement the interface independently. I'm not sure it is feasible to update all `buildAll` methods in the Kotlin plugin right now.

Reply:
See the second commit I uploaded to illustrate the said above.